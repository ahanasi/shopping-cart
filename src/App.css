.card-content {
  height: 21vh;
}
