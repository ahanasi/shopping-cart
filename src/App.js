import React, { useEffect, useState } from 'react';
import { Routes, Route } from 'react-router-dom';
import 'bulma/css/bulma.min.css';
import './App.css';
import Shop from './components/Shop';
import Cart from './components/Cart';
import Nav from './components/Nav';

function App() {
  const [cart, setCart] = useState([]);
  const [cartQuantity, setCartQuantity] = useState(0);

  const addToCart = (product, productID, productQuantity) => {
    if (productQuantity == 0) return;
    if (cart.some((item) => item.id == productID)) {
      setCart((prevState) =>
        prevState.map((product) =>
          product.id == productID
            ? {
                ...product,
                quantity: product.quantity + productQuantity,
                price: (product.unitPrice * (product.quantity + productQuantity)).toFixed(2)
              }
            : {
                ...product,
                price: (product.unitPrice * product.quantity).toFixed(2)
              }
        )
      );
    } else {
      if (productQuantity < 0) return;
      setCart((prevState) => [
        ...prevState,
        {
          id: product.id,
          title: product.title,
          unitPrice: product.price.toFixed(2),
          price: (product.price * productQuantity).toFixed(2),
          quantity: productQuantity,
          imgURL: product.image
        }
      ]);
    }
  };

  const changeItemQty = (productID, changeVal) => {
    console.log(productID, changeVal);
    const product = cart.filter((item) => item.id == productID);
    addToCart(product, productID, changeVal);
  };

  const removeItemFromCart = (productID) => {
    const newCart = cart.filter((item) => {
      return item.id != productID;
    });
    console.log(newCart);
    setCart(newCart);
  };

  useEffect(() => {
    const updateCartQuantity = () => {
      const sum = cart.reduce((a, { quantity }) => a + quantity, 0);
      setCartQuantity(sum);
    };

    updateCartQuantity();
  }, [cart]);

  return (
    <div className="App">
      <Nav cartQuantity={cartQuantity} />
      <Routes>
        <Route path="/shopping-cart" element={<Shop addToCart={addToCart} />} />
        <Route
          path="cart"
          element={
            <Cart
              cart={cart}
              removeItemFromCart={removeItemFromCart}
              changeItemQty={changeItemQty}
            />
          }
        />
      </Routes>
    </div>
  );
}

export default App;
