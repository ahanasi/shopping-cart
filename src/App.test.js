import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import App from './App';

describe('App Component', () => {
  test('full app rendering/navigating', () => {
    const { container } = render(<App />, { wrapper: MemoryRouter });
    expect(container).toMatchSnapshot();
  });
});
