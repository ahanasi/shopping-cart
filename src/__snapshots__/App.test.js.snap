// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component full app rendering/navigating 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      aria-label="main navigation"
      class="navbar"
      role="navigation"
    >
      <div
        class="navbar-brand m-2"
      >
        <a
          href="/"
        >
          <img
            class="image is-64x64"
            src="shop-logo.png"
          />
        </a>
      </div>
      <div
        class="navbar-menu"
      >
        <div
          class="navbar-end"
        >
          <div
            class="navbar-item"
          >
            <a
              href="/cart"
            >
               
              <button
                class="button is-warning"
              >
                Cart (
                0
                )
              </button>
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="columns is-multiline is-centered"
    />
  </div>
</div>
`;
